-1

Tested at: 3779f73f126148ef93fa2ed0c3cffdab1044fd82


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5b5452e63af35c538a9c49d85d2d5ee56feaa1ff
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20971/23874/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20971/23874/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20971/23874/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>gmake[1]: Target 'PostBuild' not remade because of errors.
gmake[1]: Leaving directory '/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-10-17-2300'
config/SCRAM/GMake/Makefile.rules:2088: recipe for target 'src' failed
gmake: *** [src] Error 2
gmake: Target 'all' not remade because of errors.
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5b5452e63af35c538a9c49d85d2d5ee56feaa1ff
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20971/23874/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20971/23874/git-merge-result

--- a/DQM/TrackingMonitor/python/MonitorTrackInnerTrackMuons_cff.py
+++ b/DQM/TrackingMonitor/python/MonitorTrackInnerTrackMuons_cff.py
@@ -21,13 +21,6 @@ muonInnerTrack = SimMuon.MCTruth.MuonTrackProducer_cfi.muonTrackProducer.clone()
 muonInnerTrack.muonsTag = cms.InputTag("muonsPt10")
 muonInnerTrack.selectionTags = ('All',)
 muonInnerTrack.trackType = "innerTrack"
-muonInnerTrack.TkSizeBin = 10
-muonInnerTrack.TkSizeMax = 10.
-muonInnerTrack.phiErrMax = 0.001
-muonInnerTrack.etaErrMax = 0.001
-muonInnerTrack.PVBin =  40
-muonInnerTrack.PVMin =   0.5
-muonInnerTrack.PVMax = 120.5
 
 import DQM.TrackingMonitor.TrackingMonitor_cfi
 MonitorTrackMuonsInnerTrack = DQM.TrackingMonitor.TrackingMonitor_cfi.TrackMon.clone()
@@ -48,6 +41,13 @@ MonitorTrackMuonsInnerTrack.doDCAwrt000Plots = False
 MonitorTrackMuonsInnerTrack.doSIPPlots  = True
 MonitorTrackMuonsInnerTrack.doEffFromHitPatternVsPU = True
 MonitorTrackMuonsInnerTrack.doEffFromHitPatternVsBX = False
+MonitorTrackMuonsInnerTrack.TkSizeBin = 10
+MonitorTrackMuonsInnerTrack.TkSizeMax = 10.
+MonitorTrackMuonsInnerTrack.phiErrMax = 0.001
+MonitorTrackMuonsInnerTrack.etaErrMax = 0.001
+MonitorTrackMuonsInnerTrack.PVBin =  40
+MonitorTrackMuonsInnerTrack.PVMin =   0.5
+MonitorTrackMuonsInnerTrack.PVMax = 120.5
 
 #MonitorTrackINNMuons = cms.Sequence(muonInnerTrack+MonitorTrackMuonsInnerTrack)
 MonitorTrackINNMuons = cms.Sequence(cms.ignore(muonsPt10)+muonInnerTrack+MonitorTrackMuonsInnerTrack)

--- a/DQM/TrackingMonitor/python/TrackingMonitor_cfi.py
+++ b/DQM/TrackingMonitor/python/TrackingMonitor_cfi.py
@@ -391,10 +391,6 @@ TrackMon = cms.EDAnalyzer("TrackingMonitor",
     GoodPVtxMin = cms.double( 0.),
     GoodPVtxMax = cms.double(200.),
 
-    PVBin  = cms.int32 ( 50 ),
-    PVMin  = cms.double(  0.5),
-    PVMax  = cms.double( 50.5),
-
     LUMIBin  = cms.int32 ( 4000 ),
     LUMIMin  = cms.double(    0.),
     LUMIMax  = cms.double(20000.),
@@ -414,6 +410,12 @@ LongDCABins = cms.int32(100),
 LongDCAMin = cms.double(-8.0),
 LongDCAMax = cms.double(8.0),          
 )
+
+# Overcoming the 255 arguments limit
+# TrackingRegion monitoring
+TrackMon.PVBin = cms.int32 ( 50 )
+TrackMon.PVMin = cms.double(   0.5)
+TrackMon.PVMax = cms.double( 50.5)
 
 from Configuration.Eras.Modifier_phase1Pixel_cff import phase1Pixel
 from Configuration.Eras.Modifier_phase2_tracker_cff import phase2_tracker

